src/NuGet.Core/NuGet.Build.Tasks/GenerateTemporaryTargetAssembly2.cs

```diff
@@ -0,0 +1,523 @@
+//              to compile it.
+//              
+//              The new project file will replace all the Reference Items with the 
+//              resolved ReferenctPath, add all the generated code file into Compile 
+//              Item list.
+//
+// History:
+//  05/10/05: weibz   Created.
+//  11/03/17: support NuGet PackageReference properly in temporary assembly by enabling .g.targets and .g.props to be
+//            properly included, despite the project file name change.
+//
+//---------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Collections;
+
+using System.Globalization;
+using System.Diagnostics;
+using System.Reflection;
+using System.Resources;
+using System.Xml;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.Build.BuildEngine;
+
+using MS.Utility;
+using MS.Internal.Tasks;
+
+// Since we disable PreSharp warnings in this file, PreSharp warning is unknown to C# compiler.
+// We first need to disable warnings about unknown message numbers and unknown pragmas.
+#pragma warning disable 1634, 1691
+
+namespace Microsoft.Build.Tasks.Windows
+{
+    #region GenerateTemporaryTargetAssembly2 Task class
+
+    /// <summary>
+    ///   This task is used to generate a temporary target assembly. It generates
+    ///   a temporary project file and then compile it.
+    /// 
+    ///   The generated project file is based on current project file, with below
+    ///   modification:
+    /// 
+    ///       A:  Add the generated code files (.g.cs) to Compile Item list.
+    ///       B:  Replace Reference Item list with ReferenctPath item list.
+    ///           So that it doesn't need to rerun time-consuming tatk 
+    ///           ResolveAssemblyReference (RAR) again.
+    /// 
+    /// </summary>
+    public sealed class GenerateTemporaryTargetAssembly2 : Task
+    {
+        //------------------------------------------------------
+        //
+        //  Constructors
+        //
+        //------------------------------------------------------
+
+        #region Constructors
+
+        /// <summary>
+        /// Constrcutor
+        /// </summary>
+        public GenerateTemporaryTargetAssembly2()
+        {
+        }
+
+        #endregion Constructors
+
+        //------------------------------------------------------
+        //
+        //  Public Methods
+        //
+        //------------------------------------------------------
+
+        #region Public Methods
+
+        /// <summary>
+        /// ITask Execute method
+        /// </summary>
+        /// <returns></returns>
+        public override bool Execute()
+        {
+            bool retValue = true;
+
+            // Verification
+            try
+            {
+                XmlDocument xmlProjectDoc = null;
+
+                xmlProjectDoc = new XmlDocument();
+                xmlProjectDoc.Load(CurrentProject);
```

Comment:
Why are we loading this as an XmlDocument instead of using the MSBuild construction model? If this is as it was and you're just copying it, then fine. Otherwise if it's new code, IMO we should fix this.

Reply:
Just copying the code with very minor changes.